Duvet/SheetViewController.swift

```diff
@@ -11,7 +11,15 @@ public class SheetViewController: UIViewController {
     public weak var delegate: SheetViewControllerDelegate?
 
     /// An array of the sheet items being managed by the view controller.
-    public private(set) var sheetItems = [SheetItem]()
+    public private(set) var sheetItems = [SheetItem]() {
+        didSet {
+            setNeedsStatusBarAppearanceUpdate()
+        }
+    }
```

Comment:
I'm seeing a slight flicker in the status bar when it's light and presenting a sheet that specifies a light status bar. It goes from white to black and back to white. I think `setNeedsStatusBarAppearanceUpdate()` is being called before the `currentSheetItem` is set.

I think we could drop the didSet observer here since we're calling  `setNeedsStatusBarAppearanceUpdate()` manually below as part of the transition. We could also maybe use `sheetItems.last` to get the `currentSheetItem` to avoid having to manually manage it. `sheetItems` represents the stack of sheets being displayed, with the last one currently being displayed.

Reply:
Nice catch!